mlflow/pytorch/_lightning_autolog.py

```diff
@@ -212,7 +213,7 @@ def on_train_batch_end(
         ]
         self._step_metrics.update(name for (name, _) in metric_items)
         step = trainer.global_step
-        if (step + 1) % self.log_every_n_step == 0:
+        if ((step // self._global_steps_per_training_step) + 1) % self.log_every_n_step == 0:
```

Comment:
Is this the logic that you're looking for? I wrote a very simple repro for this... 

``` python
global_steps_per_training_step = 4
log_every_n_step = 30

for z in range(500):
    if ((z//global_steps_per_training_step) + 1) % log_every_n_step == 0:
        print(f"step {z} log!")
```

which results in: 
``` plaintext
step 116 log!
step 117 log!
step 118 log!
step 119 log!
step 236 log!
step 237 log!
step 238 log!
step 239 log!
step 356 log!
step 357 log!
step 358 log!
step 359 log!
step 476 log!
step 477 log!
step 478 log!
step 479 log!
```

Which has sequential triggers that are repeated. 

Is this logic more of what you're looking for here?

``` python
for z in range(500):
    if (z % (global_steps_per_training_step * log_every_n_step) == 0) and z > 0:
        print(f"step {z} log")
```

which results in: 

```plaintext
step 120 log
step 240 log
step 360 log
step 480 log
```

Reply:
I don't think your repro matches the actual behaviour. `on_train_batch_end` is called at the end of each training batch, and within each training batch, `trainer.global_step` will have been incremented once for each optimizer. So the logic looks more like:
```python
global_steps_per_training_step = 4
log_every_n_step = 30

global_step = 0
for training_step in range(500):
    global_step += global_steps_per_training_step
    if ((global_step // global_steps_per_training_step) + 1) % log_every_n_step == 0:
        print(f"global step {global_step}, training step {training_step} log!")
```
which results in:
```
global step 116, training step 28 log!
global step 236, training step 58 log!
global step 356, training step 88 log!
global step 476, training step 118 log!
global step 596, training step 148 log!
global step 716, training step 178 log!
global step 836, training step 208 log!
global step 956, training step 238 log!
global step 1076, training step 268 log!
global step 1196, training step 298 log!
global step 1316, training step 328 log!
global step 1436, training step 358 log!
global step 1556, training step 388 log!
global step 1676, training step 418 log!
global step 1796, training step 448 log!
global step 1916, training step 478 log!
```

Reply:
ahhhh gotcha! I misread the original intention about the counter being applied to every optimizer regardless of the training iteration. This makes perfect sense!